Source/ACE.Server/Command/Handlers/DeveloperDatabaseCommands.cs

```diff
@@ -139,5 +145,119 @@ public static List<string> OutputShortcutSQLCommand(string playerName, uint char
 
             return strings;
         }
+
+        [CommandHandler("import", AccessLevel.Developer, CommandHandlerFlag.None, 1, "Imports a weenie from the Content folder", "<wcid>")]
+        public static void HandleImport(Session session, params string[] parameters)
+        {
+            var content_folder = PropertyManager.GetString("content_folder").Item;
+
+            var sep = Path.DirectorySeparatorChar;
+
+            // handle relative path
+            if (content_folder.StartsWith("."))
+            {
+                var cwd = Directory.GetCurrentDirectory() + sep;
+                content_folder = cwd + content_folder;
+            }
+
+            var di = new DirectoryInfo(content_folder);
+            if (!di.Exists)
+            {
+                SendMessage(session, $"Couldn't find content folder: {di.FullName}");
+                SendMessage(session, "To set your content folder, /modifystring content_folder <path>");
+                return;
+            }
+
+            var json_folder = $"{di.FullName}{sep}json{sep}weenies{sep}";
+
+            var wcid = parameters[0];
+            var prefix = wcid + " - ";
+
+            di = new DirectoryInfo(json_folder);
+
+            var files = di.Exists ? di.GetFiles($"{prefix}*.json") : null;
+
+            if (files == null || files.Length == 0)
+            {
+                SendMessage(session, $"Couldn't find {json_folder}{prefix}*.json");
+                return;
+            }
+
+            var fi = files[0];
+
+            // convert json -> sql
```

Comment:
You don't need this step.

You can go direct from LSD model to ACE Weenie, then add or update that weenie to the context

Reply:
You could do it that way, but that way is also less flexible and adaptable.

This was part of the whole discussion that was had awhile back. The determination was that the process should work as json->sql, and sql->json, as that is logically the model that most content editors have when working with the data.

The key here is that the .sql and .json files are always kept in tandem with each other. I update 1 or the other, and I now have representation of the data in both formats. I can then update the SQL to ACE-Patches, or the JSON to LSD, regardless of which native format I prefer to work in.

Obscuring the model conversion process, and doing it all invisibly behind-the-scenes with no .json or .sql files as the result is definitely not the way to go here

These could even be separated into standalone commands/executables that could be run outside of the server. ie, there could be json2sql and sql2json standalone progams to further make converting between the 2 models as easy as possible

Reply:
You don't go from .json to .sql

you go from .json -> LSD model -> ACE model -> (database or .sql)

Reply:
which is exactly what it's doing here

.json -> LSD model -> ACE model -> (database or .sql) literally is .json to .sql. From the perspective of the user, it's converting json to sql.

The requirements here are:

 - the .json file describing the gdle model is converted into a .sql file describing the ace model
- The user has both the .json and .sql files on their system at that point, and they can view/edit both
- The updated information is added directly to the ACE DB
- The updated information is available live in-game, ie. it clears out the existing cache for that object, so that I can spawn the item with the updated data

Reply:
I see that now. The function and var naming threw me off.